source/Windows.Devices.Gpio/Gpio​Change​Counter.cs

```diff
@@ -1,21 +1,38 @@
 ﻿//
-// Copyright (c) 2017 The nanoFramework project contributors
+// Copyright (c) 2019 The nanoFramework project contributors
 // See LICENSE file in the project root for full license information.
 //
 
+using System;
+using System.Runtime.CompilerServices;
+
 namespace Windows.Devices.Gpio
 {
     /// <summary>
     /// Counts changes of a specified polarity on a general-purpose I/O (GPIO) pin.
     /// </summary>
     /// <remarks>
-    /// <para>When the pin is an input, interrupts are used to detect pin changes. Interrupts for the pin are enabled for the specified polarity, and the count is incremented when an interrupt occurs.</para>
-    /// <para>When the pin is an output, the count will increment whenever the specified transition occurs on the pin.For example, if the pin is configured as an output and counting is enabled for rising edges, writing a 0 and then a 1 will cause the count to be incremented.</para></remarks>
-    public sealed class Gpio​Change​Counter
+    /// <para>
+    /// When the pin is an input, interrupts are used to detect pin changes unless the MCU supports a counter in hardware. 
+    /// Changes of the pin are enabled for the specified polarity, and the count is incremented when a change occurs.
+    /// </para>
+    /// <para>
+    /// When the pin is an output, the count will increment whenever the specified transition occurs on the pin. 
+    /// For example, if the pin is configured as an output and counting is enabled for rising edges, writing a 0 and then a 1 will cause the count to be incremented.
+    /// </para>
+    /// </remarks>
+    public sealed class Gpio​Change​Counter : IDisposable
     {
         // property backing fields
-        private GpioChangePolarity _Polarity = GpioChangePolarity.Falling;
+        private int _pinNumber;
+        private bool _inputMode;
+        private GpioChangePolarity _polarity = GpioChangePolarity.Falling;
+        private bool _CountActive = false;
```

Comment:
nit-pick alert 😉 the var name should be `_countActive` (mind the small caps C)

Reply:
OK i missed that one.